eclair-core/src/test/scala/fr/acinq/eclair/payment/PaymentPacketSpec.scala

```diff
@@ -177,46 +178,46 @@ class PaymentPacketSpec extends AnyFunSuite with BeforeAndAfterAll {
     assert(relay_b.expiryDelta == channelUpdate_bc.cltvExpiryDelta)
     assert(payload_b.isInstanceOf[IntermediatePayload.ChannelRelay.Standard])
 
-    val add_c = UpdateAddHtlc(randomBytes32(), 1, amount_bc, paymentHash, expiry_bc, packet_c, None)
+    val add_c = UpdateAddHtlc(randomBytes32(), 1, relay_b.amountToForward, relay_b.add.paymentHash, relay_b.outgoingCltv, packet_c, None)
     val Right(relay_c@ChannelRelayPacket(_, payload_c, packet_d)) = decrypt(add_c, priv_c.privateKey, Features(RouteBlinding -> Optional))
     assert(packet_d.payload.length == PaymentOnionCodecs.paymentOnionPayloadLength)
-    assert(relay_c.amountToForward == amount_cd)
+    assert(relay_c.amountToForward >= amount_cd)
     assert(relay_c.outgoingCltv == expiry_cd)
     assert(payload_c.outgoingChannelId == channelUpdate_cd.shortChannelId)
     assert(relay_c.relayFeeMsat == fee_c)
     assert(relay_c.expiryDelta == channelUpdate_cd.cltvExpiryDelta)
     assert(payload_c.isInstanceOf[IntermediatePayload.ChannelRelay.Blinded])
     val blinding_d = payload_c.asInstanceOf[IntermediatePayload.ChannelRelay.Blinded].nextBlinding
 
-    val add_d = UpdateAddHtlc(randomBytes32(), 2, amount_cd, paymentHash, expiry_cd, packet_d, Some(blinding_d))
+    val add_d = UpdateAddHtlc(randomBytes32(), 2, relay_c.amountToForward, relay_c.add.paymentHash, relay_c.outgoingCltv, packet_d, Some(blinding_d))
     val Right(relay_d@ChannelRelayPacket(_, payload_d, packet_e)) = decrypt(add_d, priv_d.privateKey, Features(RouteBlinding -> Optional))
     assert(packet_e.payload.length == PaymentOnionCodecs.paymentOnionPayloadLength)
-    assert(relay_d.amountToForward == amount_de)
+    assert(relay_d.amountToForward >= amount_de)
     assert(relay_d.outgoingCltv == expiry_de)
     assert(payload_d.outgoingChannelId == channelUpdate_de.shortChannelId)
     assert(relay_d.relayFeeMsat == fee_d)
     assert(relay_d.expiryDelta == channelUpdate_de.cltvExpiryDelta)
     assert(payload_d.isInstanceOf[IntermediatePayload.ChannelRelay.Blinded])
     val blinding_e = payload_d.asInstanceOf[IntermediatePayload.ChannelRelay.Blinded].nextBlinding
 
-    val add_e = UpdateAddHtlc(randomBytes32(), 2, amount_de, paymentHash, expiry_de, packet_e, Some(blinding_e))
+    val add_e = UpdateAddHtlc(randomBytes32(), 2, relay_d.amountToForward, relay_d.add.paymentHash, relay_d.outgoingCltv, packet_e, Some(blinding_e))
     val Right(FinalPacket(_, payload_e)) = decrypt(add_e, priv_e.privateKey, Features(RouteBlinding -> Optional))
     assert(payload_e.amount == finalAmount)
     assert(payload_e.totalAmount == finalAmount)
-    assert(payload_e.expiry == finalExpiry)
+    assert(add_e.cltvExpiry == finalExpiry)
```

Comment:
There is an issue here...the onion payload will contain `finalExpiry + Bolt11Invoice.DEFAULT_MIN_CLTV_EXPIRY_DELTA`. This is incompatible with the checks made in `PaymentPacket.validateBlindedFinalPayload` where we reject the payment if `add.cltvExpiry < payload.expiry`, isn't it? This issue also appear in the next tests.

I'm not sure whether it's just an issue with how the test is constructed (not correctly taking into account the blinded path's expiry) or an issue with the code itself, it's worth looking into.

Reply:
Got it, on the recipient side for Bolt 12 we only want to check that we have a few blocks before the HTLC expires, and that the HTLC expiry is lower or equal to the onion's specified expiry (am I getting this one right?).